common/smoke-test/Initialize-SmokeTests.ps1

```diff
@@ -191,30 +135,136 @@ foreach ($entry in $manifest) {
-($runManifest | ConvertTo-Json -AsArray | Set-Content -Path "$repoRoot/common/smoke-test/run-manifest.json" -Force)
+function Deploy-TestResources {
+  Param([Parameter(Mandatory = $true)] $deployManifest)
 
-Write-Verbose "Writing dependencies into Smoke Test package.json"
-$runnerPackageSpec = Get-Content "$repoRoot/common/smoke-test/package.json" | ConvertFrom-Json -AsHashtable
-$runnerPackageSpec.dependencies = $dependencies
-($runnerPackageSpec | ConvertTo-Json | Set-Content "$repoRoot/common/smoke-test/package.json")
+  $deployedServiceDirectories = @{ }
+  $baseName = 't' + (New-Guid).ToString('n').Substring(0, 16)
+  $resourceGroupName = "rg-smoke-$baseName"
+  $dependencies = New-Object 'System.Collections.Generic.Dictionary[string,string]'
+  $runManifest = @()
 
-SetEnvironmentVariable -Name "NODE_PATH" -Value "$PSScriptRoot/node_modules"
+  # Use the same resource group name that New-TestResources.ps1 generates
+  Set-EnvironmentVariable -Name 'AZURE_RESOURCEGROUP_NAME' -Value $resourceGroupName
 
-if ($CI) {
-  # If in CI mark the task as successful even if there are warnings so the
-  # pipeline execution status shows up as red or green
-  Write-Host "##vso[task.complete result=Succeeded; ]DONE"
+  $entryDeployJobs = @()
+
+  # Remove any lingering jobs from this session so output is not polluted
+  Get-Job | Stop-Job
+  Get-Job | Remove-Job
+
+  foreach ($entry in $deployManifest) {
+    if (!(Get-ChildItem -Path "$repoRoot/sdk/$($entry.ResourcesDirectory)" -Filter test-resources.json -Recurse)) {
+      Write-Verbose "Skipping $($entry.ResourcesDirectory): could not find test-resources.json"
+      continue
+    }
+
+    if ($deployedServiceDirectories.ContainsKey($entry.ResourcesDirectory) -ne $true) {
+      Write-Verbose "Starting deploy job for $($entry.ResourcesDirectory)"
+      $job = Start-NewTestResourcesJob $entry $baseName $resourceGroupName
+      $entryDeployJobs += $job
+      $deployedServiceDirectories[$entry.ResourcesDirectory] = $true;
+    }
+    else {
+      Write-Verbose "Skipping resource directory deployment (already deployed) for $($entry.ResourcesDirectory)"
+    }
+
+    Update-SamplesForService $entry
+    Update-SampleDependencies $entry $dependencies
+    $runManifest += $entry
+  }
+
+  Write-Verbose "Waiting for all deploy jobs to finish"
+  Get-Job | Wait-Job
+
+  foreach ($job in $entryDeployJobs) {
+    if ($job.State -eq [System.Management.Automation.JobState]::Failed) {
+      $errorMsg = $job.ChildJobs[0].JobStateInfo.Reason.Message
+      Write-CIWarning "Failed to deploy $($job.Name): $($errorMsg)"
+      Write-Warning "Failed to deploy $($job.Name)"
+      Write-Host $errorMsg
+      continue
+    }
+
+    Write-Verbose "setting env"
+    $deployOutput = Receive-Job -Id $job.Id
+    foreach ($key in $deployOutput.Keys) {
+      Set-EnvironmentVariable -Name $key -Value $deployOutput[$key]
+    }
+  }
+
+  # Cleanup job records
+  Get-Job | Remove-Job
```

Comment:
```suggestion
  $entryDeployJobs  | Remove-Job
```

Reply:
Might be worth doing this in a finally block.

Reply:
I added `Remove-Jobs -Force` to finally, but am not doing `Stop-Job`. From what I can tell, stop-job doesn't actually kill active jobs, it just stops the queue in case there are subsequent jobs. Since in devops the whole session will be killed, remove/stop are only for convenience with local testing, in which case I would think the developer wouldn't want to wait for all jobs to finish before starting another test run.